src/List.js

```diff
@@ -1,31 +1,53 @@
-import React from 'react';
-import { FirestoreCollection } from 'react-firestore';
+import React, { useContext } from 'react';
+import firebase from 'firebase';
 import NavLinks from './NavLinks';
-import useTokenHook from './useTokenHook';
+import { ItemsContext } from './ItemsContext';
 
 const List = () => {
-  const { token } = useTokenHook();
+  const items = useContext(ItemsContext);
+
+  const now = new Date(Date.now());
+
+  const handleChange = async e => {
+    const newData = { lastPurchased: Date.now() };
+    const db = firebase.firestore();
+
+    const itemRef = db.collection('items').doc(e.target.id);
+    itemRef.update(newData);
+  };
 
   return (
-    <FirestoreCollection
-      path="items"
-      filter={['token', '==', token]}
-      render={({ isLoading, data }) => {
-        return isLoading ? (
-          <div>loading...</div>
-        ) : (
-          <div>
-            <h1>Items</h1>
-            <ul>
-              {data.map(items => (
-                <li key={items.id}>{items.name}</li>
-              ))}
-            </ul>
-            <NavLinks />
-          </div>
-        );
-      }}
-    />
+    <div>
+      {items ? (
+        <div>
+          <h1>Items</h1>
+          <ul>
+            {items.map(item => (
+              <li key={item.id}>
+                <input
+                  type="checkbox"
+                  onChange={handleChange}
+                  id={item.id}
+                  checked={
+                    (item.lastPurchased &&
+                      now < item.lastPurchased + 1000 * 60 * 60 * 24) ||
```

Comment:
Absolutely! I would even consider going a little step further to avoid "magic numbers": You can hold the value of `1000 * 60 * 60 * 24` in a constant variable so that this comparison would ultimately read something like `now < item.lastPurchased + oneDay`, which is easier to understand when reading the code. Again, that's just a detail. Great job!